caktus_primer/templates/styleguide/styleguide-modals.html

```diff
@@ -0,0 +1,219 @@
+
+    <h2 class="styleguide-heading">Medium Modal</h2>
+    <div id="mediumDialog">
+        <button type="button" class="btn-open" id="modal_open" aria-label="open">Open Medium Modal</button>
+        <dialog>
+            <h3>Medium Modal Window</h3>
+            <p>Morbi quis nibh quam. Sed ipsum lacus, ultrices at rutrum in, dapibus in velit. Nunc dignissim, arcu in lacinia tempus, eros mi eleifend massa, ut elementum ante nulla id leo.</p>
+            <button type="button" class="btn-close" id="modal_close" aria-label="close">X</button>
+        </dialog>
+    </div>
+
+    <h2 class="styleguide-heading">Large Modal</h2>
+    <div id="largeDialog">
+        <button type="button" class="btn-open" id="modal_open" aria-label="open">Open Large Modal</button>
+        <dialog>
+            <h3>Large Modal Window</h3>
+            <p>Morbi quis nibh quam. Sed ipsum lacus, ultrices at rutrum in, dapibus in velit. Nunc dignissim, arcu in lacinia tempus, eros mi eleifend massa, ut elementum ante nulla id leo.</p>
+            <p>Phasellus ultricies varius quam. Suspendisse fringilla massa sapien, vel ornare diam bibendum nec. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.</p>
+            <button type="button" class="btn-close" id="modal_close" aria-label="close">X</button>
+        </dialog>
+    </div>
+
+    <h2 class="styleguide-heading">Modal with a Form</h2>
+        <p>Your modal can contain a variety of things, including forms!</p>
+        <div id="formDialog">
+            <button type="button" class="btn-open" id="modal_open" aria-label="open">Open Form Modal</button>
+            <dialog>
+                <h3>Modal Form Title</h3>
+                <form>
+                    <fieldset>
+                        {% for field in create_user_form %}
+                            {{ field }}
+                            {{ field.label_tag }}
+                        {% endfor %}
+                        <input type=submit value="Save" formnovalidate>
+                    </fieldset>
+                </form>
+                <button type="button" class="btn-close" id="modal_close" aria-label="close">X</button>
+            </dialog>
+        </div>
+
+    <h2 class="styleguide-heading">Usage</h2>
+    {% example wide=True %}
+        <!-- HTML -->
+        <div id="mediumDialog">
+            <button type="button" class="btn-open" id="modal_open" aria-label="open">Open Medium Modal</button>
+            <dialog>
+                <h3>Medium Modal Window</h3>
+                <p>Morbi quis nibh quam. Sed ipsum lacus, ultrices at rutrum in, dapibus in velit. Nunc dignissim, arcu in lacinia tempus, eros mi eleifend massa, ut elementum ante nulla id leo.</p>
+                <button type="button" class="btn-close" id="modal_close" aria-label="close">X</button>
+            </dialog>
+        </div>
+        {% endexample %}
+
+{% endblock %}
+
+{% block extra-js %}
+<script type="text/javascript">
```

Comment:
Why is this duplicated?

Reply:
This makes it actually work on SGP. JS integration does not work for me currently with integrating primer JS files into the app; and should be addressed separately.

Reply:
But, If that integration part doesn't work, then none of this work will actually be useable. Phil fixed a bug with the JS build in one of his branches recently, maybe that was related.

Reply:
It is usable because the code in modals.js can be exported/imported into a project. This is just so that it works on the demo site. I can try pull the latest `django-styleguide` updates; however this shouldn't be a blocker for the modal work to get done, imo.

Reply:
Sorry, we still need to get this out. The styleguide needs to demonstrate actual integration, otherwise its not acting as a "live" styleguide.

The JS integration should be working now, as well. It looks like you just needed to make the import relative and to actually export the classes from the module. I can help fix the integration, if you want? I've made the changes locally to make sure it worked.